src/main/java/io/github/mountainrange/mule/managers/ProductionManager.java

```diff
@@ -0,0 +1,201 @@
+		return Math.max(quantity - 50, 0);
+	}
+
+	/**
+	 * Get the base quantity of the given resource produced on the given terrain.
+	 * @param terrain terrain to get base production
+	 * @param resource resource to get base production
+	 * @return base production of the given resource on the given terrain
+	 */
+	private static int baseProductionOf(TerrainType terrain, ResourceType resource) {
+		return BASE_PRODUCTION.get(terrain).get(resource);
+	}
+
+	/**
+	 * Get the corresponding type of mule that produces this resource.
+	 * @param resource resource to get MuleType for
+	 * @return MuleType for the given resource
+	 */
+	public static MuleType muleProducedBy(ResourceType resource) {
+		return MULE_PRODUCED_BY.get(resource);
+	}
+
+	/**
+	 * Get the corresponding type of resource that this type of MULE produces.
+	 * @param mule MuleType to get type of resource for
+	 * @return ResourceType for the given MULE
+	 */
+	public static ResourceType producesResource(MuleType mule) {
+		return PRODUCES_RESOURCE.get(mule);
+	}
+
+	/**
+	 * Bound the given quantity by the given min and max. If quantity is less than min, return min instead. If quantity
+	 * is greater than max, return max instead. Otherwise, return quantity.
+	 * @param quantity quantity to bound
+	 * @param min lower bound
+	 * @param max upper bound
+	 * @return quantity bounded by min and max
+	 */
+	protected int bound(int quantity, int min, int max) {
+		if (min > max) {
+			String msg = String.format("Can't bound %d by %d and %d: min > max", quantity, min, max);
+			throw new IllegalArgumentException(msg);
+		}
+		return Math.min(min, Math.max(quantity, max));
+	}
+
+	static {
+		// Hard-coded base production quantities for each terrain type and resource
+		BASE_PRODUCTION = new EnumMap<>(TerrainType.class);
+
+		EnumMap<ResourceType, Integer> plainProduction = new EnumMap<>(ResourceType.class);
+		plainProduction.put(ResourceType.FOOD, 2);
+		plainProduction.put(ResourceType.ENERGY, 3);
+		plainProduction.put(ResourceType.SMITHORE, 1);
+		plainProduction.put(ResourceType.CRYSTITE, 0);
+
+		EnumMap<ResourceType, Integer> riverProduction = new EnumMap<>(ResourceType.class);
+		riverProduction.put(ResourceType.FOOD, 4);
+		riverProduction.put(ResourceType.ENERGY, 2);
+		riverProduction.put(ResourceType.SMITHORE, 0);
+		riverProduction.put(ResourceType.CRYSTITE, 0);
+
+		EnumMap<ResourceType, Integer> mountain1Production = new EnumMap<>(ResourceType.class);
+		mountain1Production.put(ResourceType.FOOD, 1);
+		mountain1Production.put(ResourceType.ENERGY, 1);
+		mountain1Production.put(ResourceType.CRYSTITE, 0);
+
+		EnumMap<ResourceType, Integer> mountain2Production = new EnumMap<>(mountain1Production);
+		EnumMap<ResourceType, Integer> mountain3Production = new EnumMap<>(mountain1Production);
+
+		mountain1Production.put(ResourceType.SMITHORE, 2);
+		mountain2Production.put(ResourceType.SMITHORE, 3);
+		mountain3Production.put(ResourceType.SMITHORE, 4);
+
```

Comment:
Thanks! I realized mountains 1, 2, and 3 were actually different when I was reading the reverse engineering guide.